### PR TITLE
Added a 'dissipating' feature for `TracedPath`

### PR DESCRIPTION
**Pull request by icedcoffeeee**
## Overview: What does this pull request change?

- Implementing #1920  into `TracedPath`
- Added `dissipating_time` parameter to note the time for the traced path to dissipate.
- Reworked `TracedPath`
- Deprecated `min_distance_to_new_point` (I didn't see it to be particularly useful)
## Motivation and Explanation: Why and how do your changes improve the library?

Code Used:
```py
class Test(Scene):
    def construct(self):
        path = NumberPlane().get_graph(lambda x: np.sin(x))
        a = Dot(path.get_start())
        b = TracedPath(a.get_center, dissipating_time=0.5, stroke_opacity=[1, 0])
        self.add(a, b)
        self.play(
            MoveAlongPath(a, path),
            run_time=5,
        )
        self.wait()
```
Output:

https://user-images.githubusercontent.com/83535735/131327705-7909c4ec-863d-440a-839a-ff8831f0671f.mp4

## Links to added or changed documentation pages
 
[TracedPath Documentation](https://manimce--1980.org.readthedocs.build/en/1980/reference/manim.mobject.changing.TracedPath.html)

## Reviewer Checklist
- [x] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [x] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Review by behackl — CHANGES_REQUESTED** [review]
Very pretty! I've left one suggestion and one additional comment that should be adressed, but other than that I think this is fine.

You could also try to add an animated test for this (the animation can just be 1sec long then, that'd be fine).

File: manim/mobject/changing.py [hunk 1]
```diff
@@ -108,35 +124,50 @@ def construct(self):
...
+                self.play(
+                    MoveAlongPath(a, path),
+                    run_time=5,
+                )
+                self.wait()
+
     """
 
+    @deprecated_params(
+        params="min_distance_to_new_point", since="v0.10.0", until="v0.11.0"
```
Comment (WampyCakes) [inline manim/mobject/changing.py:144]: Deprecation period is 2 releases, unless that has changed..

Comment (icedcoffeeee) [reply manim/mobject/changing.py:144]: Ah, I didn't know that. I looked at [this](https://github.com/ManimCommunity/manim/blob/main/manim/mobject/three_dimensions.py#L79-L84) and saw deprecation period was 1 release, so I got confused.

File: manim/mobject/changing.py [hunk 2]
```diff
@@ -108,35 +124,50 @@ def construct(self):
...
+
+        class DissipatingPathExample(Scene):
+            def construct(self):
+                path = NumberPlane().get_graph(lambda x: np.sin(x))
+                a = Dot(path.get_start())
+                b = TracedPath(a.get_center, dissipating_time=0.5, stroke_opacity=[1, 0])
+                self.add(a, b)
+                self.play(
+                    MoveAlongPath(a, path),
+                    run_time=5,
```
[the remark this answers is not in the document]
Comment (icedcoffeeee) [reply manim/mobject/changing.py]: Yup, done!

**Review by behackl — APPROVED** [review]
Thanks, this looks good to me now!

**Pull request merged**